From 8ca438b0d1256fca0a91d40196c83a79c9b680de Mon Sep 17 00:00:00 2001
From: Philippe Canal <pcanal@fnal.gov>
Date: Sat, 18 Nov 2017 00:12:34 +0100
Subject: [PATCH 25/27] Address ROOT-9114 (Issues with dictionary's enum
 forward decl)

For now, do not decorate enums, this leads to some errors, likely
due to incorrect merging.  See https://sft.its.cern.ch/jira/browse/ROOT-9114

Note: that passing the set of forward declare and then actual declaration
to clang (both official and ROOT's custom version), there is no problem
seen.
---
 lib/Interpreter/ForwardDeclPrinter.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/Interpreter/ForwardDeclPrinter.cpp b/lib/Interpreter/ForwardDeclPrinter.cpp
index 6815898f..d32cc1e5 100644
--- a/lib/Interpreter/ForwardDeclPrinter.cpp
+++ b/lib/Interpreter/ForwardDeclPrinter.cpp
@@ -342,7 +342,9 @@ namespace cling {
     if (!m_Policy.SuppressSpecifiers && D->isModulePrivate())
       Out() << "__module_private__ ";
     Out() << "enum ";
-    prettyPrintAttributes(D);
+    // For now, do not decorate enums, this leads to some errors, likely
+    // due to incorrect merging.  See https://sft.its.cern.ch/jira/browse/ROOT-9114
+    // prettyPrintAttributes(D);
     if (D->isScoped()) {
       if (D->isScopedUsingClassTag())
         Out() << "class ";
-- 
2.14.1

